tests/rest/client/v2_alpha/test_auth.py

```diff
@@ -38,11 +38,6 @@ def check_auth(self, authdict, clientip):
         return succeed(True)
 
 
-class DummyPasswordChecker(UserInteractiveAuthChecker):
-    def check_auth(self, authdict, clientip):
-        return succeed(authdict["identifier"]["user"])
```

Comment:
Is this just not used?

Reply:
It was used by https://github.com/matrix-org/synapse/pull/7438/commits/18071156e49d49847b87a418db09d04346ee9b16#diff-5fc780094ae3ec668024efd505af6a3dL170

But this checker was a patch to allow for identifiers being passed to `AuthHandler.check_auth`, but this PR removes the need for it as it providers that functionality.